SasquatchTV/SasquatchTV/ViewControllers/AnalyticsViewController.swift

```diff
@@ -3,98 +3,53 @@
-  }
-
-  @IBAction func trackPage(_ : Any) {
-    appCenter.trackPage("tvOS Page", withProperties : properties);
-  }
-
-  @IBAction func addProperty(_ : Any) {
-    let propKey : String = String(format : "key%d", properties.count + 1);
-    let propValue : String = String(format : "value%d", properties.count + 1);
-    properties.updateValue(propValue, forKey: propKey);
-    table?.reloadData();
-  }
-
-  @IBAction func deleteProperty(_ : Any) {
-    properties.removeAll();
-    table?.reloadData();
-  }
-
-  @objc func switchAnalyticsStatus(_ : Any) {
-    appCenter.setAnalyticsEnabled(serviceStatus?.selectedSegmentIndex == 0);
-    serviceStatus?.selectedSegmentIndex = appCenter.isAnalyticsEnabled() ? 0 : 1;
-  }
-  
-  @IBAction func startSession(_ sender: Any) {
-    appCenter.startSession()
-  }
+  var appCenter: AppCenterDelegate!
+  var properties : [String : String] = ["key" : "value"]
   
-  @objc func switchManualSessionTracker(_ sender: UISegmentedControl) {
-    UserDefaults.standard.set(sender.selectedSegmentIndex == 0, forKey: kMSManualSessionTracker);
-    print("Restart the app for the changes to take effect.")
-  }
-
-  //MARK: Table view data source
+  var isManualSessionTrackerEnabled = true
 
-  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-    return properties.count;
+  open override func viewDidLoad() {
+    super.viewDidLoad()
+    isManualSessionTrackerEnabled = UserDefaults.standard.bool(forKey: kMSManualSessionTracker)
+    manualSessionTrackerStatus.text = isManualSessionTrackerEnabled ? "Enabled" : "Disabled"
   }
-
-  func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-    guard let cell : MSPropertyViewCell = tableView.dequeueReusableCell(withIdentifier: "propertyViewCell", for: indexPath) as? MSPropertyViewCell else {
-      return UITableViewCell();
-    }
-    let propKey : String = Array(properties.keys)[indexPath.row];
-    cell.propertyKey?.text = propKey;
-    cell.propertyValue?.text = properties[propKey];
-    return cell;
-  }
-
-  override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-    guard let editPropertyViewController = segue.destination as? EditPropertyViewController else {
-      return;
+  
+  open override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+    guard let section = AnalyticsSections.init(rawValue : indexPath.section) else {
+      return
     }
-    guard let indexPath = table?.indexPathForSelectedRow else {
-      return;
+    switch (section) {
+    case .settings:
+      appCenter.setAnalyticsEnabled(!appCenter.isAnalyticsEnabled())
+      self.analyticsStatus.text = appCenter.isAnalyticsEnabled() ? "Enabled" : "Disabled"
+      break
+    case .trackEvent:
+      if indexPath.row == 0 {
+        appCenter.trackEvent("tvOS Event")
+      } else if indexPath.row == 1 {
+        appCenter.trackEvent("Custom tvOs Event", withProperties: properties)
```

Comment:
You use `properties` only here. Let's remove the ivar and use it inline:
appCenter.trackEvent("Custom tvOs Event", withProperties: ["key1" : "value1", "key2": "value2"])